src/main/java/tempest/ui/components/charts/BarChart.java

```diff
@@ -2,26 +2,138 @@
+    private void setupUI() {
+        this.removeAll();
+        this.add(createBarChart());
+    }
+
+    @Override
+    public void updateChart(State state) {
+        this.state = state;
+        setupUI();
+    }
+
+    /**
+     * Creates a ChartPanel containing the Bar Chart
+     *
+     * @return ChartPanel
+     */
+    private ChartPanel createBarChart() {
+        XYPlot plot = new XYPlot();
+
+        //Axis
+        plot.setDomainAxis(new DateAxis("Date"));
+        plot.setRangeAxis(new NumberAxis("Time"));
+
+        Font bold = new Font("Dialog", Font.BOLD, 15);
+        plot.getDomainAxis().setLabelFont(bold);
+        plot.getRangeAxis().setLabelFont(bold);
+
+        //Dataset
+        plot.setDataset(createDataset());
+        plot.setBackgroundPaint(Color.DARK_GRAY);
+
+        //Renderer
+        StackedXYBarRenderer renderer = new StackedXYBarRenderer();
 
-  public BarChart(State state, ViewManager<Chart> manager) {
-    super(state, manager);
-    this.add(new JLabel(getName()));
-  }
+        renderer.setBarPainter(new StandardXYBarPainter());
 
-  @Override
-  public String getName() {
-    return ChartTypes.BAR;
-  }
+        renderer.setShadowVisible(false);
+        //renderer.setMargin(0.2);
+        //renderer.setBarAlignmentFactor(0.5); //-> Doesn't work
 
-  @Override
-  public void updateChart(State state) {
-    // TODO Auto-generated method stub
+        // Hours and mins when hovering over bars
+        renderer.setDefaultToolTipGenerator((xyDataset, i, i1) -> {
+            int value = (int) xyDataset.getYValue(i, i1);
 
-  }
+            return String.format("%d hrs %02d mins", value / 60, value % 60);
+        });
 
+        plot.setRenderer(renderer);
+
+        // Creating the bar chart
+        JFreeChart chart = new JFreeChart(plot);
+        return new ChartPanel(chart);
+    }
+
+    /**
+     * @return A TableXYDataset containing the dates and length of all study sessions
+     */
+    private TableXYDataset createDataset() {
+        TimeTableXYDataset dataset = new TimeTableXYDataset();
+
+        // Sorts the study sessions so the biggest sessions are added first
+//
+//        ArrayList<StudySession> allSessions = new ArrayList<>();
+//
+//        Map<StudySession, String> dict = new HashMap<>();
+//
+//        for (Module m : state.getModules()) {
+//
+//            StudySession[] sessions = m.getStudySessions();
+//
+//            for (StudySession session : sessions) {
+//                dict.put(session, m.getName());
+//            }
+//
+//            allSessions.addAll(Arrays.asList(sessions));
+//        }
+//
+//        allSessions.sort(Collections.reverseOrder(Comparator.comparing(o -> o.duration)));
+//
+//        for (StudySession allSession : allSessions) {
+//            System.out.println(allSession.duration.toMinutes());
+//        }
+//
+//        for (StudySession s : allSessions) {
+//            dataset.add(new Day(s.date), s.duration.toMinutes(), dict.get(s));
+//        }
```

Comment:
Could this be removed too?

Reply:
Yeh removed it, although it makes more sense to have the bigger sessions at the bottom it was just really confusing having different colours in bar charts compared to the other two

Reply:
Perhaps if/when we can set the colours for all the charts then this can be added back in (is that what you're looking at doing with the chart controls)

Reply:
Should be able to do it by looping over each series in the charts and then use

```java
renderer.setSeriesPaint(seriesIndex, color);
```

I can't seem to find an easy way to get all the series in a chart though

Regardless - **_definitely not_** a priority to have bigger sessions at bottom